File: Python version/Temperature.py
import Libraries.tsys01 as tsys01
sensor = tsys01.TSYS01() # Use default I2C bus 1
sensor = tsys01.TSYS01(0) # Specify I2C bus 0
class TempSensor(object):
    def __init_(self):
        self.tempeartureF
        self.temperatureSuccess
    def getTempF(self):
        if(sensor.init()):
            temperatureSuccess = sensor.read()
            if(temperatureSuccess):
                tempeartureF = sensor.temperature(tsys01.UNITS_Farenheit)
                return tempeartureF
            else:
                print("Error Temperature")
                return 5000
        else:
            print("Failed Failed")

File: Temperature.cpp
#include <iostream>
#include "Libraries/BlueRobotics_TSYS01_Library-master/TSYS01.h" 
#include <Wire.h>

class TempSensor {
private:
    TSYS01 sensor;  
    double temperatureF;
    bool temperatureSuccess;

public:
    TempSensor(int bus = 1) {
        if (!sensor.init(bus)) {
            std::cerr << "Failed to initialize temperature sensor on I2C bus " << bus << std::endl;
        }
    }

    double getTempF() {
        if (sensor.init()) {  
            temperatureSuccess = sensor.read();  
            if (temperatureSuccess) {
                temperatureF = sensor.temperature(tsys01::UNITS_Farenheit);  // 
                return temperatureF;
            } else {
                std::cerr << "Error reading temperature" << std::endl;
                return -404; 
            }
        } else {
            std::cerr << "Sensor initialization failed" << std::endl;
            return -400;
        }
    }
};
